subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_iso.c

```diff
@@ -625,11 +625,27 @@ static void isr_tx_common(void *param,
 
 	/* Get ISO data PDU, not control subevent */
 	if (!pdu) {
-		uint8_t payload_index;
+		uint16_t payload_index;
 
-		payload_index = (lll->bn_curr - 1U) +
-				(lll->ptc_curr * lll->pto);
-		payload_count = lll->payload_count + payload_index - lll->bn;
+		if (lll->ptc_curr) {
+			uint8_t ptx_idx = lll->ptc_curr - 1; /* max. nse 5 bits */
+			uint8_t ptx_payload_idx;
+			uint16_t ptx_group_mult;
+			uint8_t ptx_group_idx;
+
+			/* Calculate group index and multiplier for deriving
+			 * pre-transmission payload index.
+			 */
+			ptx_group_idx = ptx_idx / lll->bn; /* 5 bits */
+			ptx_payload_idx = ptx_idx - ptx_group_idx * lll->bn; /* 8 bits */
+			ptx_group_mult = (ptx_group_idx + 1) * lll->pto; /* 9 bits */
+			payload_index = ptx_payload_idx + ptx_group_mult * lll->bn; /* 13 bits */
```

Comment:
The comments are somewhat helpful. Currently they are too vague (I know from context that they refer to the maximum size of the result, but that's not clear for someone else reading this later). 

As you may be aware, this is a general concern when I'm reviewing controller PRs, and in some cases there are potential issues caught and found (e.g. you've modified the type of `ptx_group_mult` and `payload_index` based on my review). 

The comments are OK assuming that the size of these values never change. A more future proof approach would be to have asserts ensuring that we don't have silent and unintentional under- or overflows.